envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/ServiceTagFilter.kt

```diff
@@ -0,0 +1,77 @@
+package pl.allegro.tech.servicemesh.envoycontrol.snapshot
+
+class ServiceTagFilter(properties: ServiceTagsProperties = ServiceTagsProperties()) {
+
+    private val tagsBlacklist: List<Regex> = properties.routingExcludedTags.map { Regex(it) }
+    private val twoTagsCombinationsByService: Map<String, List<Pair<Regex, Regex>>>
+    private val threeTagsCombinationsByService: Map<String, List<Triple<Regex, Regex, Regex>>>
+
+    init {
+        properties.allowedTagsCombinations.forEach {
```

Comment:
Could you add some UnitTests for this Filter?

Reply:
It is tested in integration test. But you are right, unit test will be useful especially for documentation purpose. I just added it.

Reply:
I also refactored the filter and moved all the service tags logic here.